### DISCUSSION TITLE
Using Dialogic 2, Unsure?

### DISCUSSION THREAD

**Discussion by BigDC** [dialogic-godot/dialogic#2180] [Q&A]
I don't know where to ask this question so I'll ask it here. I apologize if it is in the wrong place... 
I want to use Dialogic 2 for dialogue in my RPG game, should I use a new timeline each time when I interact with a different character?

**Comment by Jowan-Spooner** [comment]
Yes!
Generally timelines should be used in whatever form makes sense for your game. Depending on how much content there is for each character your approach might differ though. (E.g. each character has only a few lines vs. A character has many long interaction)

A usefull things is, that you can start a timeline at a specific label with `Dialogic.start("timeline", "label")`!

If all character interactions are relatively short, I sometimes have all interactions with a character in one timeline, each starting with a `label` (a name/title) and ending with `[end_timeline]`. 
Then I have a variable like `"next_label"` which some interactions change, so that when I interact with the character again, I can use that variable. 

Here is an example from the [test project](https://github.com/dialogic-godot/test-project/blob/main/SmallRPG/Timelines/rpg_smith.dtl) (*Note that the signal event is used to change the "next" variable, this could be done differently though*):
```ink
# First interaction with the smith
Smith: Hi, I'm Smith.
Player: Hi Smith!
Smith: I don't have time right now.
Smith: Come back later!
[signal arg="next->again"]
[end_timeline]

# Second interaction
label again
Smith: Oh, it's you again.
Player: Yep, me again.
Smith: Haha, I like you.
[signal arg="next->quest"]
[end_timeline]

# Third interaction
label quest
Smith: Alright kid...
Smith: I feel like I can trust you.
Smith: So I have a task for you.
- Okay?
	Smith: I've lost a key in the forest.
	Smith: It was in the north-east.
	Smith: Could you get it?
	- Okay!
		Smith: Thank you, kiddo.
		Player: See you later!
		[signal arg="next->quest_accepted"]
	- Argh no
		Smith: You won't? Damn.
		Smith: Well then off you go.
		[signal arg="next->quest_declined"]
- Nah.
	Smith: Not? I've judged you wrong.
	Smith: Then go!
	[signal arg="next->quest_declined"]
[end_timeline]

# Quest In Progress
label quest_declined
Smith: What <do you want/is it> now?
Smith: <Argh. I don't have time for you./ Go away!>
Smith: <Go/Go away/Buzz off>!
[end_timeline]

label quest_accepted
Smith: And?
Smith: Have you <found/ gotten> the <key/ key yet>?
- Yes [if {RPG_Example.smiths_key} > 0]
	Player: Here, I hope it's this one.
	Smith: Oh?
	Smith: Yes, that is my key!
	Smith: Thank you very much.
	Player: No problem.
	Smith: See you later then my friend.
	Player: Bye!
	[signal arg="next->end_good"]
- No
	Smith: Well, I hope you'll get it soon.
	Player: I'll try!
[end_timeline]

# Quest Finished
label end_good
Smith: It's <good/so great/wonderful> <to have/you broght> <the/my> key back.
Smith: Thank you <once again/very much/my friend>!
[end_timeline]
```

Alternatively you can simply increase a counter whenever you interact with a character and use that, but that has obvious limits. 

Or of course use a separate timeline for each encounter, which might make sense if you have few, but long interactions. 

Another note is that the visual editor get's slow (mainly the loading, the editing is fine) in timelines that have 100+ events, while the text editor handles thousands of events quite well.

I hope this helps. Good luck!
